frontend/src/reducers/user.js

```diff
@@ -0,0 +1,39 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+  login: {
+    accessToken: null,
+    userId: 0,
+    name: "",
+    statusMessage: "",
+  },
+};
+
+export const user = createSlice({
+  name: "user",
+  initialState: initialState,
+  reducers: {
+    setAccessToken: (state, action) => {
+      const { accessToken } = action.payload;
+      state.login.accessToken = accessToken;
+    },
+    setUserId: (state, action) => {
+      const { userId } = action.payload;
+      state.login.userId = userId;
+    },
+    setName: (state, action) => {
+      const { name } = action.payload;
+      state.login.name = name;
+    },
+    setStatusMessage: (state, action) => {
+      const { statusMessage } = action.payload;
+      state.login.statusMessage = statusMessage;
+    },
+    // When the logout function is called in both userProfile.js and status.js then the accessToken, userId and name are changed back to what they were in the initial state before the user was created or signed in
+    logout: (state, action) => {
+      state.login.accessToken = null;
+      state.login.userId = 0;
+      state.login.name = "";
+    }
+  },
+});
```

Comment:
A reducer is a great place to manage your user, but you need to store part of it in the local storage to make your users stay logged in after a refresh of the page. 🤓